src/Authentication/OAuth2Adapter.php

```diff
@@ -133,20 +133,22 @@ public function preAuth(Request $request, Response $response)
      */
     public function authenticate(Request $request, Response $response, MvcAuthEvent $mvcAuthEvent)
     {
-        $content       = $request->getContent();
         $oauth2request = new OAuth2Request(
-            $_GET,
-            $_POST,
+            $request->getQuery()->toArray(),
+            $request->getPost()->toArray(),
             array(),
-            $_COOKIE,
-            $_FILES,
-            $_SERVER,
-            $content,
+            ($request->getCookie() ? $request->getCookie()->getArrayCopy() : array()),
+            ($request->getFiles() ? $request->getFiles()->toArray() : array()),
+            (method_exists($request, 'getServer') ? $request->getServer()->toArray() : $_SERVER),
+            $request->getContent(),
```

Comment:
I did the above, as I was surprised we were using the superglobals; there was no reason to, and they hindered testing.